airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -267,8 +276,14 @@ private static Runnable readFromDstRunnable(final AirbyteDestination destination
             throw new DestinationException("Destination process read attempt failed", e);
           }
           if (messageOptional.isPresent()) {
-            LOGGER.info("State in DefaultReplicationWorker from destination: {}", messageOptional.get());
-            messageTracker.acceptFromDestination(messageOptional.get());
+            final AirbyteMessage message = messageOptional.get();
+            LOGGER.info("State in DefaultReplicationWorker from destination: {}", message);
+
+            messageTracker.acceptFromDestination(message);
+
+            if (message.getType() == Type.CONTROL) {
+              acceptDstControlMessage(destinationConfig, message.getControl(), connectorConfigUpdater);
```

Comment:
We may want to have a `try...catch` here, otherwise, if we fail to update the config, we'd end up failing the sync.
Same comment for the source.

Reply:
Originally I had intentionally made syncs fail if we couldn't update the config, but after thinking about this some more I agree we shouldn't fail because:

- Syncing data is our top priority and what we care about. We should keep moving data.
- These messages would be emitted for two main reasons at the moment: config migrations and handling short-lived or one-time-use tokens for oauth. 
    - For config migrations, we should be able to try again on the next sync. 
    - For the OAuth case, we could see another CONTROL message later on that might get correctly updated, or if it's really an issue we'll get an error on the next sync anyway (so we should let the current sync run its course).

The benefit of failing is we know right away something didn't go quite right, and we would want to know about these errors. However it might be better surfaced as some kind of notice rather than a full failure.

Updated to just log the error in [e7f4736](https://github.com/airbytehq/airbyte/pull/19811/commits/e7f4736e343d0ccc2fc1b8716eda30588a462fa8)